gitlab.go

```diff
@@ -993,10 +989,42 @@ func parseError(raw interface{}) string {
 	}
 }
 
-// OptionFunc can be passed to all API requests to make the API call as if you were
-// another user, provided your private token is from an administrator account.
-//
-// GitLab docs: https://docs.gitlab.com/ce/api/README.html#sudo
+// ClientOptionFunc can be used customize a new GitLab API client.
+type ClientOptionFunc func(*Client) error
```

Comment:
When extending and using the library I found it hard to find all the functions that match this signature. Would be nice to move them all to a separate file. And same with `OptionFunc`. Will also make this file smaller and easier to navigate.

Reply:
Not sure if I agree, but sure 😉 Moved the option functions to `client_options.go` and `request_options.go`.